pkgs/by-name/mi/millisecond/package.nix

```
@@ -0,0 +1,62 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  meson,
+  ninja,
+  pkg-config,
+  python313,
+  gobject-introspection,
+  gtk4,
+  desktop-file-utils,
+  appstream,
+  glib,
+  wrapGAppsHook4,
+  libadwaita,
+}:
+
+let
+  pythonEnv = python313.withPackages (p: [
+    p.pygobject3
+  ]);
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "millisecond";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "gaheldev";
+    repo = "Millisecond";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-SMGcSlbOfBX5gAwB7CaHRthf9EN5QWAN9ZzrcbQXtm8=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    ninja
+    pythonEnv
+    desktop-file-utils
+    appstream
+    glib
+    pkg-config
+    gtk4
+    libadwaita
+    gobject-introspection
+  ];
```

Most of the `buildInputs` (runtime dependencies) are actually build dependencies (`nativeBuildInputs`)

```suggestion
  nativeBuildInputs = [
    appstream
    desktop-file-utils
    glib
    gobject-introspection
    gtk4
    meson
    ninja
    pkg-config
    pythonEnv
    wrapGAppsHook4
  ];

  buildInputs = [
    gtk4 # You can omit this if you want, as libadwaita already propagates it
    libadwaita
  ];
```